test/framework/kogitoruntime.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+)
+
+// DeployRuntimeService deploy a Kogito service
+func DeployRuntimeService(namespace string, installerType InstallerType, kogitoRuntime *v1alpha1.KogitoRuntime) error {
+	GetLogger(namespace).Infof("%s deploy %s example %s with persistence %v, events %v and labels %v", installerType, kogitoRuntime.Spec.Runtime, kogitoRuntime.Name, kogitoRuntime.Spec.InfinispanMeta.InfinispanProperties.UseKogitoInfra, kogitoRuntime.Spec.KafkaMeta.KafkaProperties.UseKogitoInfra, kogitoRuntime.Spec.ServiceLabels)
+
+	switch installerType {
+	case CLIInstallerType:
+		return cliDeployRuntimeService(namespace, kogitoRuntime)
+	case CRInstallerType:
+		return crDeployRuntimeService(namespace, kogitoRuntime)
+	default:
+		panic(fmt.Errorf("Unknown installer type %s", installerType))
+	}
+}
+
+func crDeployRuntimeService(namespace string, kogitoRuntime *v1alpha1.KogitoRuntime) error {
+	if _, err := kubernetes.ResourceC(kubeClient).CreateIfNotExists(kogitoRuntime); err != nil {
+		return fmt.Errorf("Error creating example service %s: %v", kogitoRuntime.Name, err)
+	}
+	return nil
+}
+
+func cliDeployRuntimeService(namespace string, kogitoRuntime *v1alpha1.KogitoRuntime) error {
+	cmd := []string{"deploy-service", kogitoRuntime.Name}
```

Comment:
Is this already implemented? CLI does not deploy the `KogitoRuntime` CR yet, does it?

Reply:
AFAIK not yet. I guess it is a copy paste from KogitoApp.

Reply:
It's implemented, but not used. I will remove it in order to avoid misunderstandings.